packages/common/src/authentication/AuthContext.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { createContext, FC, useEffect, useMemo, useRef } from 'react';
```

Comment:
the `useState` hook, but in Context form. and then the refresh added in

Reply:
nice, it's awesome!